pkg/transformer/metrics_transformer.go

```diff
@@ -0,0 +1,215 @@
+	EventTier        string           `json:"eventTier,omitempty"`
+	Type             string           `json:"type,omitempty"`
+	EventFroze       bool             `json:"eventFroze,omitempty"`
+	EarlyAccess      EarlyAccessDTO   `json:"earlyAccess,omitempty"`
+	Prebook          PrebookDTO       `json:"prebook,omitempty"`
+	Lifecycle        LifecycleDTO     `json:"lifecycle,omitempty"`
+	EaSlots          []LifecycleDTO   `json:"eaSlots,omitempty"`
+	EventStatus      string           `json:"eventStatus,omitempty"`
+	ApplicableFees   []FeeDetailDTO   `json:"applicableFees,omitempty"`
+	DiscountingDepth string           `json:"discountingDepth,omitempty"`
+}
+
+type EarlyAccessDTO struct {
+	PointTimeMarkers []PointTimeMarkerDTO `json:"pointTimeMarkers,omitempty"`
+	DseId            string               `json:"dseId,omitempty"`
+	Description      string               `json:"description,omitempty"`
+	StartTime        int64                `json:"startTime,omitempty"`
+	EndTime          int64                `json:"endTime,omitempty"`
+	Status           string               `json:"status,omitempty"`
+}
+
+type PointTimeMarkerDTO struct {
+	Id          string          `json:"id,omitempty"`
+	EaSlots     []LifecycleDTO  `json:"eaSlots,omitempty"`
+	TimeAnchors []TimeAnchorDTO `json:"timeAnchors,omitempty"`
+}
+
+type TimeAnchorDTO struct {
+	Id        string `json:"id,omitempty"`
+	StartTime int64  `json:"startTime,omitempty"`
+}
+
+type SegmentInfoDTO struct {
+	MarketPlace   string `json:"marketPlace,omitempty"`
+	BusinessUnit  string `json:"businessUnit,omitempty"`
+	SuperCategory string `json:"superCategory,omitempty"`
+}
+
+type LifecycleDTO struct {
+	StartTime int64 `json:"startTime,omitempty"`
+	EndTime   int64 `json:"endTime,omitempty"`
+}
+
+type FeeDetailDTO struct {
+	FeeType   string  `json:"feeType,omitempty"`
+	FeeTag    string  `json:"feeTag,omitempty"`
+	FeeAmount float64 `json:"feeAmount,omitempty"`
+}
+
+type PrebookDTO struct {
+	PrebookId        string       `json:"prebookId,omitempty"`
+	PrebookLifecycle LifecycleDTO `json:"prebookLifecycle,omitempty"`
+	Name             string       `json:"name,omitempty"`
+}
+
+type EventMetricsTransformer struct {
+	Client           *http.Client
+	EventAPIEndpoint string
+}
+
+func NewEventMetricsTransformer(eventAPIEndpoint string) (*EventMetricsTransformer, error) {
+	retryClient := retryablehttp.NewClient()
+	retryClient.RetryMax = 10
+
+	client := retryClient.StandardClient()
+
+	return &EventMetricsTransformer{
+		Client:           client,
+		EventAPIEndpoint: eventAPIEndpoint,
+	}, nil
+}
+
+func (ec *EventMetricsTransformer) GetOutlierIntervalsAndInterpolate(startTime time.Time, dataPoints []metrics.DataPoint) ([]metrics.DataPoint, error) {
+	var intervals []OutlierInterval
+	fetchEventsUrl := fmt.Sprintf("http://%s/fk-event-calendar-service/v1/eventCalendar/search?startTime=%d&tier=%s&pageNo=%d&pageSize=%d", ec.EventAPIEndpoint, startTime.UnixMilli(), "ALL_MP", 0, 100)
```

Comment:
Why is the URL hardcoded here. Move it into a config.

Reply:
Also the part of integrating with a calendar API and fetching data into the generic schema can be offloaded to a different component say EventCalendarDataFetcher. That component can be DI ed into this. This will make the integration with FK specific calendar move to a different component and this generic event integration can still be used by others.